SAS/mi_mar_regression.qmd

```diff
@@ -0,0 +1,100 @@
+---
+
+title: "Multiple Imputaton: Linear Regression in SAS"
+
+---
+
+## Multiple Imputation
+
+#### Input dataset preparation before multiple imputation
+
+1.  Prepare a subset of the analysis dataset, as PROC MI requires a horizontal, one record per subject data set. More often than not, the data we impute will come from a vertical ADaM BDS data set. So we need to first transpose the aval with the avisitn as ID (assuming avisitn = 1 to 5),creating transposed variable v1-v5.
+
+```         
+data dummy;
+    length USUBJID $4;
+    do i=1 to 10;
+            sex1n=int(ranuni(0)*2);
+        do j=1 to 5;
+            USUBJID=strip(put(1000+i,best.));
+            AVISITN=j;
+            AVAL=round(1+ranuni(0),0.01);
+            if ranuni(0) <0.1 then aval=.;
+            output;
+        end;
+    end;
+    drop i j;
+run;
+proc sort data=dummy; by usubjid sex1n;run;
+proc transpose data=dummy out=dummyt(drop=_name_) prefix=v;
+    by USUBJID sex1n;
+    id avisitn;
+    var aval;
+run;
+proc print data=dummyt(obs=5);
+run;
+```
+
+```{r, echo=FALSE, fig.align='center', out.width="100%"}
+knitr::include_graphics("../images/mi_mar_linear_sas/mi_mar_reg_view_Data.PNG")
+```
+
+#### Check missing data patterns
+
+The pattern can be checked using the following code, missing data pattern could be classified as "Monotone" or "Arbitrary"
+
+```         
+ods select MissPattern;
+proc mi data=dummyt nimpute=0;
+var v1 - v5;
+run;
+```
+
+Obviously the missing pattern is arbitrary and non-monotone missing pattern.
```

Comment:
This might be my lack of stats knowledge but I don't see how this is obvious from the mi_mar_reg_missing_pattern.png output? 
Later it seems you re-use the same data for the monotone case also - is that ok?